drivers/adc/ad738x/ad738x.c

```diff
@@ -159,19 +159,13 @@ int32_t ad738x_spi_single_conversion(ad738x_dev *dev,
  *									 ONE_WIRE_MODE
  * @return 0 in case of success, negative error code otherwise.
  */
-int32_t ad738x_set_conversion_mode(ad738x_dev *dev,
-				   ad738x_conv_mode mode)
+int32_t ad738x_set_conversion_mode(struct ad738x_dev *dev,
+				   enum ad738x_conv_mode mode)
 {
-	int32_t ret;
-
-	ret = ad738x_spi_write_mask(dev,
-				    AD738X_REG_CONFIG2,
-				    AD738X_CONFIG2_SDO2_MSK,
-				    AD738X_CONFIG2_SDO2(mode));
-
-	dev->conv_mode = mode;
```

Comment:
Why drop the conv_mode parameter and assignment? if the spi_write failed the conv_mode shouldn't be updated

Reply:
I prefered using a standard driver approach:

https://github.com/analogdevicesinc/no-OS/blob/0e932e7f0bbc71baf36b781fea1dea9040d7a4aa/drivers/adc/ad738x/ad738x.c#L330

https://github.com/analogdevicesinc/no-OS/blob/0e932e7f0bbc71baf36b781fea1dea9040d7a4aa/drivers/adc/ad738x/ad738x.c#L338

Reply:
I prefered using a standard driver approach:

https://github.com/analogdevicesinc/no-OS/blob/0e932e7f0bbc71baf36b781fea1dea9040d7a4aa/drivers/adc/ad738x/ad738x.c#L330

https://github.com/analogdevicesinc/no-OS/blob/0e932e7f0bbc71baf36b781fea1dea9040d7a4aa/drivers/adc/ad738x/ad738x.c#L338